backend/layers/processing/dataset_metadata_update.py

```diff
@@ -0,0 +1,284 @@
+                val = val if isinstance(val, list) else [val]
+                seurat_metadata[seurat_metadata.names.index(key)] = StrVector(val)
+
+        base.saveRDS(rds_object, file=seurat_filename)
+
+        self.create_artifact(
+            seurat_filename,
+            DatasetArtifactType.RDS,
+            new_key_prefix,
+            new_dataset_version_id,
+            self.artifact_bucket,
+            DatasetStatusKey.RDS,
+            datasets_bucket=self.datasets_bucket,
+        )
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.RDS, DatasetConversionStatus.CONVERTED)
+
+    def update_cxg(
+        self,
+        cxg_uri: str,
+        new_cxg_dir: str,
+        dataset_version_id: DatasetVersionId,
+        metadata_update: DatasetArtifactMetadataUpdate,
+    ):
+        self.s3_provider.upload_directory(cxg_uri, new_cxg_dir)
+        ctx = tiledb.Ctx(H5ADDataFile.tile_db_ctx_config)
+        array_name = f"{new_cxg_dir}/cxg_group_metadata"
+        with tiledb.open(array_name, mode="r", ctx=ctx) as metadata_array:
+            cxg_metadata_dict = json.loads(metadata_array.meta["corpora"])
+            cxg_metadata_dict.update(metadata_update.as_dict_without_none_values())
+
+        with tiledb.open(array_name, mode="w", ctx=ctx) as metadata_array:
+            metadata_array.meta["corpora"] = json.dumps(cxg_metadata_dict)
+
+        self.business_logic.add_dataset_artifact(dataset_version_id, DatasetArtifactType.CXG, new_cxg_dir)
+        self.update_processing_status(dataset_version_id, DatasetStatusKey.CXG, DatasetConversionStatus.CONVERTED)
+
+    def update_metadata(
+        self,
+        old_dataset_version_id: DatasetVersionId,
+        new_dataset_version_id: DatasetVersionId,
+        metadata_update: DatasetArtifactMetadataUpdate,
+    ):
+        old_dataset_version = self.business_logic.get_dataset_version(old_dataset_version_id)
+        if old_dataset_version.status.processing_status != DatasetProcessingStatus.SUCCESS:
+            self.logger.info(
+                f"Dataset {old_dataset_version_id} is not successfully processed. Skipping metadata update."
+            )
+            return
+
+        artifact_uris = {artifact.type: artifact.uri for artifact in old_dataset_version.artifacts}
+
+        if DatasetArtifactType.RAW_H5AD in artifact_uris:
+            raw_h5ad_uri = artifact_uris[DatasetArtifactType.RAW_H5AD]
+        else:
+            self.logger.error(f"Cannot find raw H5AD artifact uri for {old_dataset_version_id}.")
+            raise ValueError
+
+        self.process(new_dataset_version_id, raw_h5ad_uri, self.artifact_bucket)
+
+        new_artifact_key_prefix = self.get_key_prefix(new_dataset_version_id.id)
+
+        with multiprocessing.Pool() as pool:
+            if DatasetArtifactType.H5AD in artifact_uris:
+                self.logger.info("Main: Starting thread for h5ad update")
+                pool.starmap_async(
```

Comment:
I think `apply_async` over `starmap_async` would be better suited , since there is not iterating over multiple jobs and doesn't block

Reply:
This is true but also the results are not being awaited anywhere; statuses will be marked as success/failure before the subprocesses complete. [My fix for it](https://github.com/chanzuckerberg/single-cell-data-portal/pull/6274).